tests/plugins/test_fetch_sources.py

```diff
@@ -0,0 +1,247 @@
+"""
+Copyright (c) 2019 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+
+from __future__ import absolute_import
+
+import os
+from textwrap import dedent
+
+import koji
+import pytest
+from flexmock import flexmock
+
+from atomic_reactor import util
+from atomic_reactor.constants import (
+    PLUGIN_BUILD_ORCHESTRATE_KEY,
+    PLUGIN_FETCH_SOURCES_KEY)
+from atomic_reactor.inner import DockerBuildWorkflow
+from atomic_reactor.plugin import PreBuildPluginsRunner, PluginFailedException
+from atomic_reactor.plugins.pre_fetch_sources import FetchSourcesPlugin
+from atomic_reactor.plugins.pre_reactor_config import (ReactorConfigPlugin,
+                                                       WORKSPACE_CONF_KEY, ReactorConfig)
+from tests.constants import TEST_IMAGE
+from tests.stubs import StubInsideBuilder, StubSource
+
+
+KOJI_HUB = 'http://koji.com/hub'
+KOJI_ROOT = 'http://koji.localhost/kojiroot'
+KOJI_UPLOAD_TEST_WORKDIR = 'temp_workdir'
+KOJI_BUILD = {'build_id': 1, 'nvr': 'foobar-1-1', 'name': 'foobar', 'version': 1, 'release': 1}
+
+DEFAULT_SIGNING_INTENT = 'empty'
+
+
+def mock_reactor_config(workflow, tmpdir, data=None, default_si=DEFAULT_SIGNING_INTENT):
+    if data is None:
+        data = dedent("""\
+            version: 1
+            odcs:
+               signing_intents:
+               - name: invalid
+                 keys: ['notUsed']
+               - name: one
+                 keys: ['usedKey']
+               - name: multiple
+                 keys: ['notUsed', 'usedKey', 'notUsed2']
+               - name: unsigned
+                 keys: ['']
+               - name: empty
+                 keys: []
+               default_signing_intent: {}
+               api_url: invalid
+               auth:
+                   ssl_certs_dir: {}
+            """.format(default_si, tmpdir))
+
+    workflow.plugin_workspace[ReactorConfigPlugin.key] = {}
+
+    config = {}
+    if data:
+        tmpdir.join('cert').write('')
+        config = util.read_yaml(data, 'schemas/config.json')
+
+    workflow.plugin_workspace[ReactorConfigPlugin.key][WORKSPACE_CONF_KEY] = ReactorConfig(config)
+
+
+def mock_dockerfile(tmpdir):
```

Comment:
I'm curious about reason why this plugin needs Dockerfile

Reply:
Thanks for spotting this. This remained after I removed the Dockerfile dependency at some point.